WeatherAPI/Startup.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using WeatherBL;
+using WeatherBL.Interfaces;
+using WeatherBL.Validators;
+using WeatherDAL;
+using WeatherDAL.Models;
+
+namespace WeatherAPI
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+
+            services.AddControllers();
+
+            services.AddScoped<HttpClient>();
+            services.AddScoped(opt => Configuration.GetSection("Config").Get<ConfigOptions>());
+            services.AddScoped<IValidator, CityValidator>();
+            services.AddScoped<IWeatherProvider, WeatherProvider>();
+            services.AddScoped<IWeatherService, WeatherService>();
+
+            services.AddSwaggerGen(c =>
+            {
+                c.SwaggerDoc("v1", new OpenApiInfo { Title = "WeatherAPI", Version = "v1" });
+            });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+                app.UseSwagger();
+                app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "WeatherAPI v1"));
+            }
+
+            app.UseRouting();
+
+            app.UseAuthorization();
```

Comment:
What is the reason for use it here?

Reply:
app.userouting() is used to route requests on middleware
app.useauthorization() is used to authorizes a user to access secure resources

Reply:
Do we have secured resources here?

Reply:
no cause I didn't add any kind of authentication or authorization